contracts/pangolin-periphery/PangolinBridgeMigrationRouter.sol

```diff
@@ -0,0 +1,358 @@
+pragma solidity ^0.7.6;
+
+import "../pangolin-core/interfaces/IPangolinERC20.sol";
+import "../pangolin-lib/libraries/TransferHelper.sol";
+import "./interfaces/IBridgeToken.sol";
+import "./libraries/Roles.sol";
+import "./libraries/PangolinLibrary.sol";
+import "hardhat/console.sol";
+
+contract PangolinBridgeMigrationRouter {
+    using SafeMath for uint;
+    using Roles for Roles.Role;
+
+    Roles.Role private adminRole;
+    mapping(address => address) public bridgeMigrator;
+
+    constructor() public {
+        adminRole.add(msg.sender);
+    }
+
+    // safety measure to prevent clear front-running by delayed block
+    modifier ensure(uint deadline) {
+        require(deadline >= block.timestamp, 'PangolinBridgeMigrationRouter: EXPIRED');
```

Comment:
I'm not sure how is this supposed to prevent front running. Seem easily bypassable.

Reply:
This ensures a tx is only executable within so long of submitting. Avoids:
* malicious node got this tx and decided to hold onto it while other tx is confirmed (easily avoided by submitting frontrun tx ordered earlier in the same block sure)
* innocent ISP or network latency causes the tx to execute later than expected

Reply:
what @bmino, this isn't supposed to be a silverbullet for the problem but rather a amendment to the situation, also, this is unlikely to be the case on Avalanche, but since this is a fork of uniswap, we should keep it consistent with those checks

Reply:
Resolving, no action to be done here

Reply:
Ok thanks for the explanation. I guess this check will **detect (and revert)** rather than **prevent** possible front running , which was what confused me. I.e. you can't actually stop front running from happening. Suggest to amend the comment to clarify.